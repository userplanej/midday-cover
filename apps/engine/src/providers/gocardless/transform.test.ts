import { expect, test } from "bun:test";
import {
  transformAccount,
  transformAccountBalance,
  transformTransaction,
} from "./transform";

test("Transform income transaction", () => {
  expect(
    transformTransaction({
      transactionId:
        "8wwecA0PsWWCLPLazQzht2oOcxSIQ5UlWPHaNBpC3tjYtc002faJcjWzRpyO4sjz66kRpb7_5rA",
      entryReference: "5490990006",
      bookingDate: "2024-02-23",
      valueDate: "2024-02-23",
      transactionAmount: {
        amount: "-38000.00",
        currency: "SEK",
      },
      additionalInformation: "LÃ–N         160434262327",
      proprietaryBankTransactionCode: "Transfer",
      internalTransactionId: "86b1bc36e6a6d2a5dee8ff7138920255",
    }),
  ).toMatchSnapshot();
});

test("Transform accounts", () => {
  expect(
    transformAccount({
      id: "b11e5627-cac8-41c9-a74a-2b88438fe07d",
      created: "2024-02-23T13:29:47.314568Z",
      last_accessed: "2024-03-06T16:34:16.782598Z",
      iban: "3133",
      institution_id: "PLEO_PLEODK00",
      status: "READY",
      owner_name: "",
      account: {
        resourceId: "3133",
        currency: "SEK",
        name: "Pleo Account",
        product: "Pleo",
        cashAccountType: "TRAN",
        iban: "123",
        ownerName: "Name",
      },
      balance: {
        currency: "SEK",
        amount: "1942682.86",
      },
      institution: {
        id: "PLEO_PLEODK00",
        name: "Pleo",
        bic: "PLEODK00",
        transaction_total_days: "90",
        countries: ["DK", "GB", "DE", "SE", "ES", "IE", "DK"],
        logo: "https://cdn-logos.gocardless.com/ais/PLEO_PLEODK00.png",
      },
    }),
  ).toMatchSnapshot();
});

test("Transform account balance", () => {
  expect(
    transformAccountBalance({
      currency: "SEK",
      amount: "1942682.86",
    }),
  ).toMatchSnapshot();
});
