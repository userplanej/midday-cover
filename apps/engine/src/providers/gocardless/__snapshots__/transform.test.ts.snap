// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform income transaction 1`] = `
{
  "amount": -38000,
  "balance": null,
  "category": "transfer",
  "currency": "SEK",
  "currency_rate": null,
  "currency_source": null,
  "date": "2024-02-23",
  "description": null,
  "id": "86b1bc36e6a6d2a5dee8ff7138920255",
  "method": "transfer",
  "name": "LÃ¶n 160434262327",
  "status": "posted",
}
`;

exports[`Transform accounts 1`] = `
{
  "balance": {
    "amount": 1942682.86,
    "currency": "SEK",
  },
  "currency": "SEK",
  "enrollment_id": null,
  "id": "b11e5627-cac8-41c9-a74a-2b88438fe07d",
  "institution": {
    "id": "PLEO_PLEODK00",
    "logo": "https://cdn-engine.midday.ai/PLEO_PLEODK00.png",
    "name": "Pleo",
    "provider": "gocardless",
  },
  "name": "Pleo Account",
  "type": "depository",
}
`;

exports[`Transform account balance 1`] = `
{
  "amount": 1942682.86,
  "currency": "SEK",
}
`;
